miner/worker.go

```diff
@@ -613,6 +618,31 @@ func (w *worker) resultLoop() {
 	}
 }
 
+// timestampLoop is a loop that updates the timestamp used for blocks when it
+// is stale by more than a certain threshold.
+// TODO: Re-think this as everything comes together more.
+func (w *worker) timestampLoop() {
+	timer := time.NewTimer(0)
```

Comment:
Is this meant to only run once or on an interval? Right now, it will only run once since a timer represents a single event in the future. See: https://gobyexample.com/timers

If its meant to run on an interval, this pattern can be used:

```go
// run once per minute
for now := range time.Tick(time.Minute) {
  ...
}
```

Reply:
The interval will change depending on the state when the timer is triggered. You can see it being reset [here](https://github.com/ethereum-optimism/go-ethereum/pull/14/files#diff-83814d75e37ea3b4eb16ffc7202295d3R635) and [here](https://github.com/ethereum-optimism/go-ethereum/pull/14/files#diff-83814d75e37ea3b4eb16ffc7202295d3R635)

Reply:
Ah you are right, the call to `timer.Reset`. Thank you for clarifying!